contrib/FastPFOR-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+set(FASTPFOR_LIBRARY_DIR ${CMAKE_SOURCE_DIR}/contrib/FastPFOR)
+
+set (SRCS
+    "${FASTPFOR_LIBRARY_DIR}/src/bitpacking.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/bitpackingaligned.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/bitpackingunaligned.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/codecfactory.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/horizontalbitpacking.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/simdbitpacking.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/simdunalignedbitpacking.cpp"
+    "${FASTPFOR_LIBRARY_DIR}/src/streamvbyte.c"
+    "${FASTPFOR_LIBRARY_DIR}/src/varintdecode.c"
+)
+
+add_library(_fastpfor ${SRCS})
+
+target_include_directories(_fastpfor SYSTEM BEFORE PUBLIC "${FASTPFOR_LIBRARY_DIR}/headers/")
+
+if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
+    # SIMDe (SIMD everywhere) is a dependency of FastPFOR only on ARM machines.
```

Comment:
Reminder for myself: In case of non-x86 / non-ARM, we should either disable the inverted index (throw an exception) or fallback to a non-SIMD implementation (if available).